### PR TITLE
ban command finished

### PR DESCRIPTION
**Pull request by arjunshankkar**

### CODE REVIEW

File: src/commands/ban.js [hunk 1]
```diff
@@ -2,9 +2,42 @@ const Discord = require("discord.js");
...
+    if(!message.member.hasPermission('MANAGE_MESSAGES')){ //check permissions
+        return errors.noPerms(message, "Moderator Role Needed")
+    }
+
+    args = message.content.split(' ').slice(1);
+    const userToBan = message.mentions.users.first();
+    const banReason = args.slice(1).join(' ');
+    
+    if (userToBan) {
+        if (userToBan.length > 2) {
```
Comment (Ulramok) [inline src/commands/ban.js]: message.mentions.users.first() returns one user object. So you can't use the length to see if only one member is being mentioned.

File: src/commands/ban.js [hunk 2]
```diff
@@ -2,9 +2,42 @@ const Discord = require("discord.js");
...
+            return message.channel.send(`${message.author} Invalid format. Please ban one user at a time.`);
+        }
+        else {
+            if (message.author === userToBan) {
+                return message.channel.send(`${message.author} You cannot ban yourself.`);
+            }
+            if (!banReason) {
```
Comment (Ulramok) [inline src/commands/ban.js]: Let's create a default ban reason rather than forcing them to specify a reason to ban.

File: src/commands/ban.js [hunk 3]
```diff
@@ -2,9 +2,42 @@ const Discord = require("discord.js");
...
+                return message.channel.send(`${message.author} You must have a reason to ban!`);
+            }
+            else {
+                await message.guild.ban(userToBan) // Bans the user
+                const banConfirmationEmbed = new Discord.RichEmbed()
+                .setColor('RED')
+                .setDescription(`✅ ${userToBan} has been successfully banned!`);
```
Comment (Ulramok) [inline src/commands/ban.js]: This message should delete itself after a set amount of time. Don't think there is a need to keep receipts in this channel. We can create a log feature that maintains a list of banned users, edited/deleted messages and etc.

File: src/commands/ban.js [hunk 4]
```diff
@@ -2,9 +2,42 @@ const Discord = require("discord.js");
...
+                message.channel.send({
+                embed: banConfirmationEmbed
+                });
+            }
+        }
+    } else { 
+        return message.channel.send(`${message.author} Invalid format. Please put the user to be banned wrapped in double quotes, i.e. ?ban @thisUser#2131 Spam`);
```
Comment (Ulramok) [inline src/commands/ban.js]: So you should not place the user to be banned in double quotes but just ping them. The reason to ban can be in double quotes. It's easier to just check if the message mentions someone. Which I think is what you are doing atm.

File: src/commands/ban.js (hunk 2 above)
Comment (arjunshankkar) [reply src/commands/ban.js]: will do

File: src/commands/ban.js (hunk 4 above)
Comment (arjunshankkar) [reply src/commands/ban.js]: yeah the message is inaccurate, ill fix this

File: src/commands/ban.js (hunk 1 above)
Comment (arjunshankkar) [reply src/commands/ban.js]: youre right, ill remove this check

**Pull request merged**